mesonbuild/modules/unstable_wayland.py

```diff
@@ -12,24 +12,47 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
 import os
+import typing as T
+
+from mesonbuild.programs import OverrideProgram
 
 from . import ExtensionModule, ModuleReturnValue
-from ..build import CustomTarget
+from ..build import CustomTarget, Executable
 from ..interpreter.type_checking import NoneType, in_set_validator
 from ..interpreterbase import FeatureNew, typed_pos_args, typed_kwargs, KwargInfo
 from ..mesonlib import File, MesonException
 
+if T.TYPE_CHECKING:
+    from typing_extensions import Literal,TypedDict
+
+    from . import ModuleState
+    from ..dependencies import Dependency
+    from ..interpreter import Interpreter
+    from ..programs import ExternalProgram
+    from ..mesonlib import FileOrString
+
+    class ScanXML(TypedDict):
+
+        public: bool
+        client: bool
+        server: bool
+
+    class FindProtocol(TypedDict):
+
+        state: Literal['stable', 'staging', 'unstable']
+        version: T.Optional[int]
 
 class WaylandModule(ExtensionModule):
 
     @FeatureNew('wayland module', '0.62.0')
-    def __init__(self, interpreter):
+    def __init__(self, interpreter: Interpreter) -> None:
         super().__init__(interpreter)
 
-        self.protocols_dep = None
-        self.pkgdatadir = None
-        self.scanner_bin = None
+        self.protocols_dep: T.Optional[Dependency] = None
+        self.pkgdatadir: T.Optional[str] = None
+        self.scanner_bin: T.Optional[T.Union[ExternalProgram, OverrideProgram, Executable]] = None
```

Comment:
You don't need to mention OverrideProgram, since it's a type of ExternalProgram anyway.